### PR TITLE
Plugin support packages and tooling

### PR DESCRIPTION
**Pull request by azdagron**
- SDK interfaces
- protoc-gen-go-spire tool for generating SPIRE stubs conforming to the SDK interfaces
- Plugin loading and test packages
- Makefile for doing codegen

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+	@echo "$(bold)Usage:$(reset) make $(cyan)<target>$(reset)"
+	@echo "  $(cyan)generate$(reset)                              - generate protocol buffers and plugin interface code"
+	@echo "  $(cyan)generate-check$(reset)                        - ensure generated code is up to date"
+	@echo "  $(cyan)protogen$(reset)                              - compile protocol buffers"
+	@echo "  $(cyan)protogen-check$(reset)                        - ensure generated protocol buffers are up to date"
+	@echo "  $(cyan)plugingen$(reset)                             - generate plugin interface code"
+	@echo "  $(cyan)plugingen-check$(reset)                       - ensure generated plugin interface code is up to date"
+	@echo
+	@echo "For verbose output set V=1"
+	@echo "  for example: $(cyan)make V=1 build$(reset)"
```
Comment (amartinezfayo) [inline Makefile:26]: Since `build` is not a target in this Makefile, we may mention a valid target here.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+golangci_lint_version = v1.27.0
+golangci_lint_dir = $(build_dir)/golangci_lint/$(golangci_lint_version)
+golangci_lint_bin = $(golangci_lint_dir)/golangci-lint
+
+protoc_version = 3.14.0
+ifeq ($(arch1),aarch64)
+protoc_url = https://github.com/protocolbuffers/protobuf/releases/download/v$(protoc_version)/protoc-$(protoc_version)-$(os2)-aarch_64.zip
+else
+endif
```
Comment (amartinezfayo) [inline Makefile:94]: It seems that this else should not be empty
```suggestion
protoc_url = https://github.com/protocolbuffers/protobuf/releases/download/v$(protoc_version)/protoc-$(protoc_version)-$(os2)-$(arch1).zip
endif
```

**Review by amartinezfayo — COMMENTED** [review]
Looking good.

File: Makefile (hunk 1 above)
Comment (azdagron) [reply Makefile:26]: :+1, also cleaned up some of those targets that don't exist

File: Makefile (hunk 2 above)
Comment (azdagron) [reply Makefile:94]: Wow, I don't know how that happened :laughing:

**Review by amartinezfayo — APPROVED** [review]
Thank you for all this @azdagron, looks good to me!

**Comment by azdagron** [comment]
Thanks, @amartinezfayo !

**Pull request merged**